pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -802,6 +826,17 @@ func (lrw *logicalReplicationWriterProcessor) flushBuffer(
 		stats.Add(lrw.bhStats[i])
 	}
 
+	if stats.debatchResolvedErrs > 0 {
+		// de-batching helped -- up the backoff.
+		lrw.skipBatchingStep = max(1, min(lrw.skipBatchingStep*2, 100))
+		lrw.skipBatching += lrw.skipBatchingStep
```

Comment:
We are doing this here and then again at 838 below. Is that intentional?

Reply:
nope; I went through a slightly messy rebase to pull the first refactor commit out so it could join the backport party without the behavior changes.